### ISSUE TITLE
Provide alternative display formats for dates in Data Explorer

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#4538]
Brought up by @weshoke: "When dates are rendered, it would be really nice if they were in a format that could be input easily into a query"

Two possible options to implement this:
* Display both the nicely formatted datum as well as a format that's easy to paste into ReQL next to each other:
```
Fri Jan 02 1970 10:17:36 GMT+00:00 | r.epochTime(123456)
```
* Provide a configuration setting to change the date format between different options.

**Comment by weshoke** [comment]
regarding the formatting, I was even thinking of something where you don't necessarily write a valid query expression like `r.epochTime(123456)` but have the option of printing an ISO8601 format that you could then paste as an argument to `r.ISO8601`.  I could imagine a few options:

- current rendering, which is the JS Date object's toString() method
- print as ISO8601
- print as epochTime

**Comment by weshoke** [comment]
Also, giving a preference to render dates as ISO8601 strings would make the tree view produce valid JSON, which is not the case currently

**Comment by weshoke** [comment]
I'm using I'm using Chrome `Version 43.0.2357.132 (64-bit)`  on OSX and pasting from the Tree View into the text box above it.

**Comment by weshoke** [comment]
I have the same problem with Safari.  I'm also using RethinkDB 2.0.1

**Comment by danielmewes** [comment]
> Also, giving a preference to render dates as ISO8601 strings would make the tree view produce valid JSON, which is not the case currently

:+1:

**Comment by bnosrat** [comment]
It seems kind of risky and potentially confusing to have a query output something different in the web ui as opposed to the cli or repl.

**Comment by danielmewes** [comment]
@bnosrat I agree that we shouldn't deviate too much, but I think it's ok in this case. We could keep the default setting the same.

**Comment by deontologician** [comment]
Another option would be to have what you copy be a serialized version, and
the display be a pretty version. I'm not 100% sure, but I think there's
already two layers right now, which is why you can copy/paste the tree view

**Comment by danielmewes** [comment]
Ah, I'm not sure right now if that would be really confusing or really nice. Have to think about it more.

**Comment by weshoke** [comment]
@bnosrat The problem for me is that the way the JS Date objet is handled makes using the results as JSON impossible.

`r.expr({date: r.epochTime(10000)})`

When rendered in the Data Explorer gives:
```
{
"date": Thu Jan 01 1970 02:46:40 GMT+00:00
}
```

When you pass the result through `JSON.stringify(res, null, 2)` you get:

``` {"date":"1970-01-01T02:46:40.000Z"}```

which is actually useful.  I understand the need not to confuse people, but I would think that the result being actual usable JSON would be an equal if not bigger priority.  It's so incredibly useful and productive from a user perspective to be able to take the results from the Data Explorer and paste them into other places that can handle JSON.  As it stands, the data shown in the Data Explorer can't be used anywhere else without going through by hand and replacing the dates with proper strings.

**Comment by marshall007** [comment]
There are several cases where "Tree view" produces invalid JSON (URLs are another example). I think it would make more sense to keep tree view pretty printing as-is and have these setting available under "Raw view". Copying from there is always going to be more reliable anyway.

**Comment by weshoke** [comment]
That sounds like a good solution.  Another possible one is some kind of "copy as JSON" command/button.  Basically, if there's some way I can grab the data easily as JSON I would be extremely happy.

**Comment by weshoke** [comment]
so, let me update my thoughts... reconsidering the raw view, it's is valid JSON

```
{
    "date": {
        "$reql_type$": "TIME",
        "epoch_time": 10000,
        "timezone": "+00:00"
    }
}
```

thinking about the workflow some more, I guess if the Dates were strings, you'd lose the semantic tag of the field being a Date as opposed to an ordinary string. I can see the value of this and in any tooling that takes the data directly, it would just need an interpreting layer to convert `$reql_type$`to a JS Date object.  

Looking at it from this perspective, I'm not sure my feature request is really a useful one.
